### PR TITLE
Movie Randomizer by Darya, React-native app 

### PR DESCRIPTION
**Pull request by DALA746**
https://expo.dev/@/projects/movie-randomizer

### CODE REVIEW

File: components/Movies.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	const generateMovie = () => {
+		fetch(FETCH_URL)
+			.then((res) => res.json())
+			.then((json) => {
+				setMovieList(json.results);
+				//setLoading(true)
+			})
+			.finally(() => setLoading(false))
+	};
+
```
Comment (nehrwein) [inline components/Movies.js:42]: By commenting out the loading in line 38, there's no use of the ActivityIndicator at all in this component. If I remember right from your demo, there was some interference with your other code, if you put the setLoading(true) in line 34 where it belongs. So maybe delete it altogether, if you don't have use for it or try to fix the issue.

File: components/Movies.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      });
+      if (result.action === Share.sharedAction) {
+        if (result.activityType) {
+          // shared with activity type of result.activityType
+        } else {
+          // shared
+        }
+      } else if (result.action === Share.dismissedAction) {
+        // dismissed
+      }
```
Comment (nehrwein) [inline components/Movies.js:57]: Interessting function! Why do you need all the if/else-statements, if there's no code inside?

File: components/Movies.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+			.then((json) => {
+				setMovieList(json.results);
+				//setLoading(true)
+			})
+			.finally(() => setLoading(false))
+	};
+
+	// share function, that gives possibility to share directly from your phone and send a message with movie title 
+	const onShare = async () => {
+    try {
```
Comment (nehrwein) [inline components/Movies.js:45]: What does 'try' do? 🙂

File: components/Movies.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        // dismissed
+      }
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+	if(loading) {
+		return <ActivityIndicator />
+	};
```
Comment (nehrwein) [inline components/Movies.js:65]: as stated above: loading is always on false

File: utils/urls.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const API_KEY = '963cfc8ae0b11289f364cb1de6574bcf'
+
+export const FETCH_URL = `https://api.themoviedb.org/3/movie/top_rated?api_key=${API_KEY}&language=en-US&page=2`;
```
Comment (nehrwein) [inline utils/urls.js:3]: Great, that you created an extra file to store the API-Url and the API-key!

**Review by nehrwein — COMMENTED** [review]
Congrats to a great project, Darya! Your app as seen in the demo looks and performs amazing and your code is clean and well-structured. You seem to have gained a good understanding of React Native this week and have incorporated cool features in your app like the swipe-navigation and the sharing-function.
Really an impressive and inspring project! 🎥 ❤️ 👍

File: README.md [hunk 6]
```diff
@@ -1,13 +1,9 @@
 # Project React Native App 📱
 
-Replace this readme with your own information about your project.
+This week, I build my first React Native App that shows a random movie from an API and possibilty to share your choice with friends and family. Build with React Native and Styled Componenets.
```
[the remark this answers is not in the document]
Comment (DALA746) [reply README.md:4]: #relatable :)

File: components/Movies.js (hunk 1 above)
Comment (DALA746) [reply components/Movies.js:42]: Yes! Thanks for the note! :)

File: .expo-shared/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+> Why do I have a folder named ".expo-shared" in my project?
```
Comment (Rosanna86) [inline .expo-shared/README.md:1]: Great with explanations

File: App.js [hunk 8]
```diff
@@ -1,25 +1,37 @@
...
 
 const App = () => {
+
 	return (
-		<Container>
-			<Title>This is your cool app!</Title>
-			<Title>Go to App.js and start coding</Title>
-			<Title>💅💅💅</Title>
-		</Container>
+		<NavigationContainer>
```
Comment (Rosanna86) [inline App.js:13]: Great navigation. And nice that the swipe was built in. I began with the tabBarBottom and that one didn't have swipe function. So I changed :)

File: components/Movies.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+										size={20}
+										color="yellow"
+									/>
+									{randomMovie.vote_average}
+								</Rate>
+								<Description>
+									{randomMovie.overview}
+								</Description>
+
+								<ButtonContainer>
```
Comment (Rosanna86) [inline components/Movies.js:92]: This inspired me - thank you!

**Review by Rosanna86 — COMMENTED** [review]
Easy code to read, great project, good with comments! Great app Darya!

**Pull request closed without merging**